src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -214,17 +228,37 @@ def add_filesystem(self, spec):
             }
         }
 
-        try:
-            self.rook_api_post(
-                "cephfilesystems/",
-                body=rook_fs
-            )
-        except ApiException as e:
-            if e.status == 409:
-                log.info("CephFilesystem '{0}' already exists".format(spec.name))
-                # Idempotent, succeed.
-            else:
-                raise
+        with self.ignore_409("CephFilesystem '{0}' already exists".format(spec.name)):
+            self.rook_api_post("cephfilesystems/", body=rook_fs)
+
+    def add_nfsgw(self, spec):
+        # TODO use spec.placement
+        # TODO use spec.min_size (and use max_size meaningfully)
+        # TODO warn if spec.extended has entries we don't kow how
+        #      to action.
+
+        rook_nfsgw = {
+            "apiVersion": ROOK_API_NAME,
+            "kind": "CephNFS",
+            "metadata": {
+                "name": spec.name,
```

Comment:
Is this name going to be chosen by the higher-level abstraction, like the export volume module?

Reply:
Yes. It's a name for the cluster of NFS servers.